dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/events/event/JdbcEventStore.java

```diff
@@ -145,6 +150,8 @@
 @Repository( "org.hisp.dhis.dxf2.events.event.EventStore" )
 public class JdbcEventStore implements EventStore
 {
+    private static final String RELATIONSHIP_IDS_QUERY = " left join (select ri.programstageinstanceid as ri_psi_id, json_agg(r.relationshipid) as psi_rl FROM relationshipitem ri \n"
+        + " left join relationship r on ri.relationshipid = r.relationshipid GROUP by ri_psi_id)  as fgh on fgh.ri_psi_id=event.psi_id ";
```

Comment:
Do we need this second join? We already have the relationshipid from relationshipitem ri

Reply:
resolved